app/controllers/verify/finance_controller.rb

```diff
@@ -17,20 +19,39 @@ def create
 
     private
 
+    def step
+      @_step ||= Idv::FinancialsStep.new(
+        idv_finance_form: idv_finance_form,
+        idv_session: idv_session
+      )
+    end
+
+    def step_params
+      params.require(:idv_finance_form).permit(:finance_type, *Idv::FinanceForm::FINANCE_TYPES)
+    end
+
+    def confirm_step_needed
+      redirect_to verify_phone_path if idv_session.financials_confirmation.try(:success?)
+    end
+
     def render_form
-      if finance_params[:finance_type] == 'ccn'
+      if step_params[:finance_type] == 'ccn'
         render :new
       else
         render 'verify/finance_other/new'
       end
     end
 
-    def idv_finance_form
-      @_idv_finance_form ||= Idv::FinanceForm.new(idv_session.params)
+    def track_idv_event
+      result = {
+        success: idv_session.financials_confirmation.success?
+      }
+
+      analytics.track_event(Analytics::IDV_FINANCE_CONFIRMATION, result)
```

Comment:
I still see this as one form. While there are two different types of validations, they must both pass in order for the form submission to be considered successful. To the user, they are submitting one form, and receiving feedback about whether or not they can continue to the next step. 

Because the two validations are required, they need to be grouped together IMO. Just because one validation is being processed externally doesn't mean it has to be a separate step. By making it a separate step, it makes it look like one can fail while the other can pass, which is not the case.

A similar example is our email validation. We have 2 types of validations: one for the format of the email, and one to validate the MX value of the domain name (which won't allow `test@example.com` for example. We use the `valid_email` gem for that, which makes an API call for the MX validation, which we don't control. Both the format and the MX value must be valid for the form to go through, but we don't have separate steps for them. They're all validations applied to the same form object.